internal/services/servicebus/servicebus_namespace_customer_managed_key_resource.go

```diff
@@ -0,0 +1,265 @@
+}
+
+func (r ServiceBusNamespaceCustomerManagedKeyResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"namespace_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: namespaces.ValidateNamespaceID,
+		},
+		"key_vault_key_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: keyVaultValidate.NestedItemIdWithOptionalVersion,
+		},
+		"identity_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: commonids.ValidateUserAssignedIdentityID,
+		},
+		"infrastructure_encryption_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			ForceNew: true,
+		},
+	}
+}
+
+func (r ServiceBusNamespaceCustomerManagedKeyResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ServiceBusNamespaceCustomerManagedKeyResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ServiceBus.NamespacesClient
+			var cmk ServiceBusNamespaceCustomerManagedKeyModel
+
+			if err := metadata.Decode(&cmk); err != nil {
+				return err
+			}
+
+			id, err := namespaces.ParseNamespaceID(cmk.NamespaceID)
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("%s was not found", *id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if resp.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` is nil", *id)
+			}
+
+			if resp.Model.Properties != nil && resp.Model.Properties.Encryption != nil && resp.Model.Properties.Encryption.KeyVaultProperties != nil && len(*resp.Model.Properties.Encryption.KeyVaultProperties) > 0 {
+				return metadata.ResourceRequiresImport(r.ResourceType(), *id)
+			}
+
+			payload := resp.Model
+
+			keyId, err := keyVaultParse.ParseOptionallyVersionedNestedItemID(cmk.KeyVaultKeyID)
+			if err != nil {
+				return err
+			}
+
+			payload.Properties.Encryption = &namespaces.Encryption{
+				RequireInfrastructureEncryption: pointer.To(cmk.InfrastructureEncryptionEnabled),
+				KeySource:                       pointer.To(namespaces.KeySourceMicrosoftPointKeyVault),
+				KeyVaultProperties: &[]namespaces.KeyVaultProperties{
+					{
+						KeyName:     pointer.To(keyId.Name),
+						KeyVersion:  pointer.To(keyId.Version),
+						KeyVaultUri: pointer.To(keyId.KeyVaultBaseUrl),
+					},
+				},
+			}
+
+			if cmk.IdentityID != "" {
+				identityId, err := commonids.ParseUserAssignedIdentityID(cmk.IdentityID)
+				if err != nil {
+					return err
+				}
+				(*payload.Properties.Encryption.KeyVaultProperties)[0].Identity = &namespaces.UserAssignedIdentityProperties{
+					UserAssignedIdentity: pointer.To(identityId.ID()),
```

Comment:
The validation function on the property will ensure that the user input is a user assigned identity ID with correct casing in the static segments, so there isn't strictly a need to parse that here if we're just passing the string value of it to the API

```suggestion
				(*payload.Properties.Encryption.KeyVaultProperties)[0].Identity = &namespaces.UserAssignedIdentityProperties{
					UserAssignedIdentity: pointer.To(cmk.IdentityID),
```